kernel/include/ksched.h

```diff
@@ -13,6 +13,16 @@
 #include <tracing/tracing.h>
 #include <stdbool.h>
 
+/* WARNING WARNING WARNING HERE THERE BE DRAGONS
+ *
+ * Many of the APIs here are unsafe to call without sched_spinlock held.
+ * This is particularly the case with most of the APIs here which take a
+ * struct k_thread pointer argument.
+ *
+ * The contents of this file should be treated as private to the core kernel
+ * and may change at any time.
```

Comment:
Is this really needed?  That's the point of these include files being under kernel/ and not include/ to begin with.  They all filled with internal APIs shared only among files in the kernel directory.

Reply:
All changes in this file are from #29668; the discussion yesterday led to an agreement to move what are ultimately going to be public API functions into this file with a `z_` prefix so I can use them in the internal work implementation.  I didn't bother trying to edit descriptions or implementation from there.

Reply:
@pabigot This is about the comment. This file was always private and we have many like those. The dragon warning is really not needed, I know where this is coming from.

Reply:
The comment was placed in this file in #29668, and I left it there.  I'll remove it next push.